rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java

```diff
@@ -453,13 +435,66 @@ private void selectNetworkInterface() {
         if (rmURL != null) {
             try {
                 logger.debug("Detecting a network interface to bind the node");
-                String networkInterface = RMConnection.getNetworkInterfaceFor(rmURL);
-                logger.info("Node will be bounded to the following network interface " + networkInterface);
-                CentralPAPropertyRepository.PA_NET_INTERFACE.setValue(networkInterface);
-            } catch (Exception e) {
-                logger.debug("Unable to detect the network interface", e);
+                InetAddress ipAddress = getNetworkInterfaceFor(rmURL);
+                if (ipAddress != null) {
+                    logger.info("Node will be bounded to the following ip address " + ipAddress);
+                } else {
+                    throw new IllegalStateException("No valid IP address found, exiting.");
+                }
+
+            } catch (URISyntaxException e) {
+                throw new IllegalStateException("Unable to detect the network interface", e);
+            }
+        }
+    }
+
+    /**
+     * Selects a network interface for proactive runtime based on provided url.
+     *
+     * @return the IP address associated with the selected network interface or null if no interface would be found
+     * @throws URISyntaxException    if the given string violates RFC&nbsp;2396,
+     * @throws IllegalStateException if no suitable address was found
+     */
+    public static InetAddress getNetworkInterfaceFor(String url) throws URISyntaxException, IllegalStateException {
+
+        if (url != null && CentralPAPropertyRepository.PA_NET_INTERFACE.getValue() == null &&
+                CentralPAPropertyRepository.PA_NET_NETMASK.getValue() == null && CentralPAPropertyRepository.PA_HOSTNAME != null) {
+            int port;  // use default port for rmi
+            String host;
+            URI parsedUrl = new URI(url);
+            if (parsedUrl.getScheme().equals("pamr")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "pamr" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)